src/nvim/api/ui_events.in.h

```diff
@@ -11,79 +11,79 @@
 #include "nvim/ui.h"
 
 void mode_info_set(Boolean enabled, Array cursor_styles)
-  FUNC_API_SINCE(3);
+  FUNC_API_SINCE(3) FUNC_API_REDRAW;
```

Comment:
can you explain why `FUNC_API_REDRAW` is needed? Why not generate handlers for everything in `ui_events.in.h` and just have empty implementations for some of them?

Reply:
If `FUNC_API_REDRAW` is not used, then the functions to be implemented would have to be explicitly defined, moreover some methods like `popupmenu_select` even though are in `ui_events.in.h` are not at all required as it is mapped to `wildmenu_select`. So in order to avoid such nuances, I decided to add that tag.

Reply:
What we can do later on is to instead put a tag on the events that are _not_ implemented in the client, to convey the same information with less tags (as client should support almost all events, essentially everything but the events we already have deprecated).

Reply:
Sure. I'll do it.